src/common/adc.h

```diff
@@ -3,17 +3,57 @@
 #include "inttypes.h"
 #include "stm32f4xx_hal.h"
 #include "config_buddy_2209_02.h"
+#include "main.h"
 
-const uint8_t CHANNEL_NOZZLE = 0;
-const uint8_t CHANNEL_BED = 1;
-const uint8_t CHANNEL_TEMP_2 = 2;
-const uint8_t CHANNEL_PINDA = 3;
-const uint8_t CHANNEL_BED_MON = 4;
+extern ADC_HandleTypeDef hadc1;
 
-extern void adc_dma_init(ADC_HandleTypeDef *adc);
-extern void adc_dma_deinit(ADC_HandleTypeDef *adc);
-extern uint16_t get_adc_channel_value(ADC_HandleTypeDef *adc, uint32_t channel);
+template <ADC_HandleTypeDef &adc, size_t channels>
+class AdcDma {
+public:
+    AdcDma()
+        : m_data() {};
+    void init() {
+#ifndef SIM_HEATER
+        HAL_ADC_Init(&adc);
+        HAL_ADC_Start_DMA(&adc, (uint32_t *)m_data, channels); // Start ADC in DMA mode and
```

Comment:
Suggest: Use c++ style cast (`static_cast`) instead of c-style cast.

Reply:
Solved